src/overlays/actors/ovl_Obj_Lightswitch/z_obj_lightswitch.c

```diff
@@ -389,126 +353,88 @@ void ObjLightswitch_Update(Actor* thisx, GlobalContext* globalCtx) {
     tempPos.z = this->actor.world.pos.z;
-
     gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-
-    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80960BC4[this->faceState]));
-
-    gSPDisplayList(POLY_OPA_DISP++, 0x6000260);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(sLightswitchFaceGfx[this->faceState]));
+    gSPDisplayList(POLY_OPA_DISP++, D_06000260);
 
     tempRot.x = this->actor.shape.rot.x;
     tempRot.y = this->actor.shape.rot.y;
     tempRot.z = this->actor.shape.rot.z + this->edgeRot;
-    SysMatrix_SetStateRotationAndTranslation(tempPos.z, tempPos.y, tempPos.x, &tempRot);
-    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
-
+    SysMatrix_SetStateRotationAndTranslation(tempPos.x, tempPos.y, tempPos.z, &tempRot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
     gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-
-    gSPDisplayList(POLY_OPA_DISP++, 0x6000398);
+    gSPDisplayList(POLY_OPA_DISP++, D_06000398);
 
     tempRot.z = this->actor.shape.rot.z - this->edgeRot;
-    SysMatrix_SetStateRotationAndTranslation(tempPos.z, tempPos.y, tempPos.x, &tempRot);
-    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
-
+    SysMatrix_SetStateRotationAndTranslation(tempPos.x, tempPos.y, tempPos.z, &tempRot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
     gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-
-    gSPDisplayList(POLY_OPA_DISP++, 0x6000408);
+    gSPDisplayList(POLY_OPA_DISP++, D_06000408);
 
     CLOSE_DISPS(globalCtx->state.gfxCtx);
 }
-#else
-void ObjLightSwitch_DrawOpa(ObjLightswitch* this, GlobalContext* globalCtx);
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Lightswitch_0x8095FBF0/ObjLightSwitch_DrawOpa.asm")
-#endif
 
-#if NON_EQUIVELENT
-// same issue as above
 void ObjLightSwitch_DrawXlu(ObjLightswitch* this, GlobalContext* globalCtx) {
+    s32 pad;
     Vec3f tempPos;
     Vec3s tempRot;
 
     OPEN_DISPS(globalCtx->state.gfxCtx);
 
     func_8012C2DC(globalCtx->state.gfxCtx);
-
-    // temp_v0->words.w1 = ((((s32) this->colorB >> 6) & 0xFF) << 8)
-    //| (((s32) this->colorR >> 6) << 0x18)
-    //| ((((s32) this->colorG >> 6) & 0xFF) << 0x10)
-    //| (((s32) this->colorAlpha >> 6) & 0xFF);
-    gDPSetEnvColor(POLY_XLU_DISP++, this->colorR >> 6, this->colorG >> 6, this->colorB >> 6, this->colorAlpha >> 6);
-
-    gSPSegment(POLY_XLU_DISP++, 0x09, D_801AEF88);
+    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(this->colorR >> 6), (u8)(this->colorG >> 6), (0, (u8)(this->colorB >> 6)), (u8)(this->colorAlpha >> 6));
+    gSPSegment(POLY_XLU_DISP++, 0x09, &D_801AEF88);
```

Comment:
That was all? I forgot to cast to u8 and the two were out of order?

That's kinda disappointing... surprised I didn't try that.

Reply:
The (u8) helped but there were still ordering issues which the permuter found the meme match `(0, (u8)(this->colorB >> 6))`. I also had to use the `&` for `D_801AEF88` and there was an x-z swap later on in the function.

Reply:
oh I didnt even see the `(0, value)` interesting